theseus/optimizer/linear/tests/test_lu_cuda_sparse_solver.py

```diff
@@ -60,8 +56,108 @@ def test_sparse_solver():
         b = linearization.b[i].cpu()
         atb = torch.Tensor(csrAi.transpose() @ b)
 
-        # the linear system solved is with matrix (AtA + damping*I)
-        atb_check = ata @ solved_x[i].cpu()  # + damping * solved_x[i].cpu()
+        # the linear system solved is with matrix AtA
+        atb_check = ata @ solved_x[i].cpu()
 
         max_offset = torch.norm(atb - atb_check, p=float("inf"))
         assert max_offset < 1e-4
+
+
+def check_sparse_solver_multistep(test_exception: bool):
+
+    if not torch.cuda.is_available():
+        return
+
+    num_steps = 3
+    torch.manual_seed(37)
+
+    void_objective = th.Objective()
+    void_ordering = th.VariableOrdering(void_objective, default_order=False)
+    solver = th.LUCudaSparseSolver(
+        void_objective,
+        linearization_kwargs={"ordering": void_ordering},
+        num_solver_contexts=(num_steps - 1) if test_exception else num_steps,
+    )
+    linearization = solver.linearization
+
+    batch_size = 4
+    void_objective._batch_size = batch_size
+    num_rows, num_cols, data, col_ind, row_ptr = _build_sparse_mat(batch_size)
+    linearization.num_rows = num_rows
+    linearization.num_cols = num_cols
+    linearization.A_col_ind = col_ind
+    linearization.A_row_ptr = row_ptr
+
+    # Only need this line for the test since the objective is a mock
+    solver.reset(batch_size=batch_size)
+
+    As = [
+        torch.randn((batch_size, len(col_ind)), dtype=torch.double).cuda()
+        for _ in range(num_steps)
+    ]
+    bs = [
+        torch.randn((batch_size, num_rows), dtype=torch.double).cuda()
+        for _ in range(num_steps)
+    ]
+    c = torch.randn((batch_size, num_cols), dtype=torch.double).cuda()
+
+    def batched_dot(a, b):
+        return torch.sum(a * b, dim=1)
+
+    def iterate_solver(As, bs):
```

Comment:
Would be helpful to add some comments with a high level summary of what this does. If I understand correctly, this returns

```
y = dot(z,c)
z = sum(A_i \ b_i)
```

Reply:
will add comments yeah